server/channels/app/support_packet.go

```diff
@@ -19,61 +15,86 @@ import (
 	"github.com/mattermost/mattermost/server/public/model"
 	"github.com/mattermost/mattermost/server/public/shared/mlog"
 	"github.com/mattermost/mattermost/server/public/shared/request"
-	"github.com/mattermost/mattermost/server/v8/config"
-)
-
-const (
-	cpuProfileDuration = 5 * time.Second
 )
 
 func (a *App) GenerateSupportPacket(c request.CTX, options *model.SupportPacketOptions) []model.FileData {
-	// If any errors we come across within this function, we will log it in a warning.txt file so that we know why certain files did not get produced if any
-	var warnings []string
-
-	// Creating an array of files that we are going to be adding to our zip file
-	fileDatas := []model.FileData{}
-
 	// A array of the functions that we can iterate through since they all have the same return value
 	functions := map[string]func(c request.CTX) (*model.FileData, error){
-		"support package": a.generateSupportPacketYaml,
-		"plugins":         a.createPluginsFile,
-		"config":          a.createSanitizedConfigFile,
-		"cpu profile":     a.createCPUProfile,
-		"heap profile":    a.createHeapProfile,
-		"goroutines":      a.createGoroutineProfile,
-		"metadata":        a.createSupportPacketMetadata,
+		"support packet": a.generateSupportPacketYaml,
+		"plugins":        a.createPluginsFile,
+		"config":         a.createSanitizedConfigFile,
+		"cpu profile":    a.createCPUProfile,
+		"heap profile":   a.createHeapProfile,
```

Comment:
Maybe we can use platform methods here directly.

Reply:
The platform methods do not accept a `request.CTX` and hence can't be used directly.

Reply:
Should we introduce `request.CTX` into the `GossipClient` ?

Reply:
I think we should, the point of that context is to carry all the information per request.

Reply:
I've updated the code via ad465caa3b to use the platform method directly. Please let me know what you think.

Reply:
Looks good, thanks!